### ISSUE TITLE
Which echo models can find HA Bridge?

### ISSUE THREAD

**Issue by heikoh81** [bwssytems/ha-bridge#890]
Hi there,

I'm using HA Bridge with 2 Echo Dots, 2 Echos (2017) and one Echo Show.
As official Hue seems to move to cloud only (and Google already connects to Hue-cloud only), I'm wondering what Echos have the capability to discover a Hue Bridge on the LAN only.

So far I know that
- Echo Show can no longer discovery local Hue-Bridges (or Emulators) on ist own.
- Echo Dot (bought 2017) can discover local Hue Bridges

What about
- Echo (2017)?
- Echo Plus (2017) (advertised as a smart home hub...)?
- Echo Spot?

Please post your experience.
Maybe this Information can than be added to the Project page of HA Bridge.

Greetings,
Heiko

**Comment by mopby** [comment]
echo spot has problems too

**Comment by michaelhinchey** [comment]
Ive had No problems yet with the echo with built in smart hub. Running 5.1.

**Comment by bwssytems** [comment]
So here is my list:
Works:
Echo/Dot 1st Gen

Issues:
Echo Show - Does not have local upnp discovery at all
Echo/Dot 2nd Gen - seems to not finish upnp discovery
Software Alexa Voice Service - Does not have local upnp discovery at all

Not Sure:
Echo Plus
Echo Spot

**Comment by michaelhinchey** [comment]
I have no issues with echo plus discorer ing the bridge as I stated earlier.

**Comment by bwssytems** [comment]
@michaelhinchey Thanks!

**Comment by ml9907** [comment]
In my case Echo dot 2nd gen (2017) can discover all ha-bridge devices.

**Comment by audiofreak9** [comment]
I have a Spot, I'll try it alone tonight and report.

**Comment by audiofreak9** [comment]
Using 5.2.0RC13 on RPi2 B+ wired lan connection, here are my test results:

1. I created a new ha-bridge device (exec type) named "Spot", with only the Echo Spot powered up, it discovered my new device.

2. I created a new ha-bridge device (exec type) named "Show", with only the Echo Show powered up, it discovered my new device.

Until now, I have never tried discovery with a Spot or Show alone, I have multiple Dots (1 & 2) and a Gen 1 Echo.  I have always been able to use the Show to discover devices (with this mix) since September.  Got the Spot in December it has always discovered for me as well.  Don't really want to do a forget all and attempt full discovery, I have over 100 devices with multiple groups.

**Comment by heikoh81** [comment]
Thanks for all your replies.
I'm not sure where Amazon (and Hue) are going with their policy to drop local discovery on newer devices like the Show - although this was advertised as a key feature. What if there are no more dots available to buy or the function is disabled by firmware update? Then hours of work settings up my HA Bridge configuration (with 116 virtual devices) would be lost...
On the contrary, the Echo Plus still seems to support local discovery.

@bwssytems 
I think my echo dot is 2nd gen, as it was bought in May 2017?
- I also may have experienced UPNP timing problems during discovery, as sometimes it finds all my devices, sometimes not.
- I always have to "forget all devices" on the Alexa website, then do several discoveries after adding 1 new device in HA Bridge. Otherwise, the new device and the total number of devices is displayed correctly on the Alexa website, however, the voice command always replies with "device not found" (although displayed correctly on the Alexa website). 
- I noticed that "renumbering devices" in HA Bridge after forgetting all devices on the Alexa website often helps to solve voice commands.

**Comment by michaelhinchey** [comment]
I would imagine in a few yrs the software would be outdated because Alexa
is growing in the way it interacts.  However I'm sure the developer of this
will make it useful no matter.









    Every dream is a goal with a date on it; if you can conceive it believe
it, you can achieve it. Don't let anyone steal your dream. Make the
decision to take a positive attitude, take pride in yourself and what you
do, work hard and stay with it. A mind is a terrible thing to waste.


"I felt bad because I couldn't wheelie; until I met a man with no bicycle"

"You are young at any age... if you are planning on tomorrow"

**Comment by audiofreak9** [comment]
@heikoh81 

> I'm not sure where Amazon (and Hue) are going with their policy to drop local discovery on newer devices

My 2 cents, I could be wrong, but I _think_ the Philips Hue bridge gen 1 only supports local discovery I don't own one.  If this is the case, I don't see local discovery ending until Philips officially ends support for their gen 1 bridge.

**Comment by andocromn** [comment]
I have a 2nd Gen Dot and an Amazon Tap.  both work fine

**Comment by bwssytems** [comment]
I have a feeling that there may be software updates that may have changed the behavior for the better on 2nd Gen echo's as it seems more success is being reported here.

**Comment by riphill** [comment]
Well I spent a very long time failing at discovery, slowly learning about the various settings of ha-bridge - I effectively gave up in the end.

I moved over to other methods but had the strange issue where the App and the Web GUI would discover my devices, but not my Echo.  It some vain attempt to sort the issue I contacted Amazon support who walked me through resetting everything I'd achieved in terms of setup.  Not expecting anything to come of it, I reset the Echo Dot (2nd Gen) and re-ran discovery via voice and to my astonishment it all worked first time - Philips Hue, ha-bridge and another skill I'd enabled.  I felt like I've lost 2 days of my life, but it all works now with ha-bridge-5.2.0RC19.jar on a raspberry pi (Debian Jessie), but for ages it just looked broken.  I could see M-Search, discovery.xml calls, upnp discovery template 1,2 and 3 and request for Hue lights then... nothing.

Incidentally the dot was registered to my partner's Amazon account rather than my own, and I'm intending to register it back and hope it all doesn't break...

So in short... success with an Echo Dot 2nd Gen after a full reset.

**Comment by michaelhinchey** [comment]
You may have to set all of it up again. Apps and everything. Let us know
how it all goes.









    Every dream is a goal with a date on it; if you can conceive it believe
it, you can achieve it. Don't let anyone steal your dream. Make the
decision to take a positive attitude, take pride in yourself and what you
do, work hard and stay with it. A mind is a terrible thing to waste.


"I felt bad because I couldn't wheelie; until I met a man with no bicycle"

"You are young at any age... if you are planning on tomorrow"

**Comment by heikoh81** [comment]
@riphill 
Thanks for this hint.

I had kind of a strange experience myself recently.
I bought a Netvue 720p IP Camera (36€) to seamlessly work as a baby monitor with my Echo Show (would prefer to use standard IP cams though).
I added the Netvue Skill (my first and only skill at all), said "Alexa find devices", and out of a sudden, the search initiated on the Echo Show found the correct number of devices (125 = 124 HABridge + 1 Netvue IP Cam).
Initiating search on the Show has NEVER worked before!

It seems - how odd it may appear - that adding an skill now has optimized the local search...

**Comment by riphill** [comment]
I have an idea of what I think has happened now.  The Echo Dot was registered to my partner's Amazon account.  I ran discovery in the web app logged in as me and the devices would appear but not be recognised by the Echo Dot so it wouldn't turn them on.  It would appear since the devices were already "discovered" it wouldn't add them again - however the Dot seems to retain some device information.  When I reset the Echo it was joined to my account and any voice discovery was reflected in my web/phone app.  Any groups I added in the app were also picked up by the Dot.

Now my partner has stolen the Dot back it won't recognise any of the devices, and I can't ask it to perform actions about the devices I've discovered or groups created on my account.  Since I don't have access to my partner's Amazon login I can't see what it's discovered, but I know what they're called so I just have to live with that for the moment.  I think because the Philips App is linked to my account it isn't picking up the scenes either - so this does seem to be an issue to be aware of.

**Comment by heikoh81** [comment]
@riphill:
However, all my Echos have always been associated with my own Amazon account, and still there are/were the discovery issues. So I'm not sure if your situation is an explanation in general.

**Comment by bwssytems** [comment]
tested my new 2nd gen echo by itself and it finds devices properly. I also use the security setting to 'Use Link Button' and select link before discover and that is working. This maybe an expectation of the 2nd gen echo.

**Comment by heikoh81** [comment]
Where can I find that 'Use Link Button'.
Can't find it on 5.2.0.

**Comment by bwssytems** [comment]
@heikoh81 It is under the security settings button on the bridge control tab

**Comment by mopby** [comment]
I still have problems with echo spot. Is there any update on how to set things up? I did a fresh install of 5.2.1 and have enabled "trace upnp calls" and set the send delay to 650. I also enabled "use link button" under security settings. The spot still doesnt find anything unfortunately. I've spent hours to figure things out so far

I have the german version of the spot btw.

**Comment by heikoh81** [comment]
I just did a device discovery from Alexa website (German user).
Right after hitting "Find devices", I activated "use link button".
Discovery was perfect on the first run.

I, however, also have 1 Echo Show, 2 Echo 2017, 2 Echo Dot 2017 in my Network.
HA Bridge has 126 HTTP-GET-Devices.
No Echo Spot.

**Comment by mopby** [comment]
Thank you for the quick answer. With my echo dot 2017 running in the same network I can discover all devices as well. I read in another thread that it should be possible to discover devices standalone with the spot. The reason is that I want to use my spot in another flat. But I think I'll have to buy another one.

**Comment by hargathor** [comment]
Hello everyone,

I have an Echo Plus [(FR Amazon Link)](https://www.amazon.fr/gp/product/B0794XQK5S/ref=oh_aui_detailpage_o00_s00?ie=UTF8&psc=1) and it couldn't discover any device.
I've done the 2 steps discovery using the link button. I've also tried forgetting all my devices but no luck...
I have a ha-bridge `ha-bridge-5.2.2RC2` deployed on a Raspberry PI2 on the 8088 port with a Apache2 redirect for `/api*` (as explained in the documentation)

What am I missing ? Is anyone had any luck ?

**Comment by blacktiger333** [comment]
Having problems with discovering my devices in HA Bridge 5.2.1 since about 2 weeks. I'm using Amazon Echo Dot 3 gen, Echo Spot, and Echo Dot 2 gen. With my Echo Dot 2 gen  I can discover my devices from pimatic (after I delete them from the alexa list this works every time I try it), but no devices where discovered with HA Bride with any of my alexa devices since 2 weeks. Any idea? Has anyone the same problem?

**Comment by bwssytems** [comment]
including in the community information https://github.com/bwssytems/ha-bridge/wiki/Informational-Items-Collected-from-the-Community#q-which-echo-models-can-find-ha-bridge
